src/Hint/Context.hs

```diff
@@ -128,7 +125,7 @@ addPhantomModule mod_text =
                    (old_top, old_imps) <- runGhc getContext
                    --
                    runGhc $ GHC.addTarget t
-                   res <- runGhc $ GHC.load (GHC.LoadUpTo m)
+                   res <- runGhc $ GHC.loadPhantomModule m
```

Comment:
I don't understand. With ghc-9.4, `loadPhantomModule m` is defined as `GHC.load GHC.LoadAllTargets`, which does not mention `m`. So where does `m` get loaded?

Reply:
Also, [this comment](https://github.com/haskell-hint/hint/pull/152#issuecomment-1537520685) says that calling `getContext` and then `setContext` does not work anymore, because the saved context refers to the modules using some internal IDs which are no longer valid after calling `LoadAllTargets`. So why is the code still using `getContext` and `setContext`? Is the comment wrong, and `setContext` does work after all?

Reply:
I now think that `setContext` does fail, but then the caller fixes it.